dev/deps/spark-deps-hadoop-3-hive-2.3

```diff
@@ -225,7 +225,7 @@ parquet-jackson/1.13.1//parquet-jackson-1.13.1.jar
 pickle/1.3//pickle-1.3.jar
 py4j/0.10.9.7//py4j-0.10.9.7.jar
 remotetea-oncrpc/1.1.2//remotetea-oncrpc-1.1.2.jar
-rocksdbjni/8.3.2//rocksdbjni-8.3.2.jar
+rocksdbjni/8.5.3//rocksdbjni-8.5.3.jar
```

Comment:
Unfortunately, Apache Spark 3.5.0 RC5 is also using RocksDB 8.3.2.

Reply:
Maybe, we need to change the issue type of SPARK-45110. Currently, it's filed as `Improvement` for Spark 4.0.0.

Reply:
Done
<img width="659" alt="image" src="https://github.com/apache/spark/assets/15246973/8dbc1822-43e4-4d76-95aa-a7e9f31b4a4f">